### ISSUE TITLE
Limit Order Architecture Design Spike

### ISSUE THREAD

**Issue by 0xean** [shapeshift/web#7905]
Brief spike to outline state management, slice, reusable components, any suggestion on refactors that we might need ahead of implementation. 

< .5 days

Outline findings in gist or on this ticket and we can review as a team ahead of implementation

**Comment by woodenfurniture** [comment]
### Initial refactoring

**TL;DR - move swapper specific stuff to higher order components**

The following changes are likely required to prepare the `TradeInput` component and its children for reuse with limit orders:

- Compose the existing `TradeInputHeader`, `TradeInputBody`, `ConfirmSummary` components without `Acknowledgement` wrappers and `CollapsibleQuoteList`, and pass all feature-specific callbacks and data in as props.
- General cleanup - Move functions like `getQuoteErrorTranslation` to a shared utilities file.
- Within `TradeInputHeader`, `TradeInputBody`, `ConfirmSummary` components, replace `useAppSelector` and `useAppDispatch` calls with props where state or logic is not shared.
- Likely have to make the "cog" menu button contents (on the top-right) a prop too, since limit orders will have different items.

### General architecture

**TL;DR - use the same exact approach as arbitrum claims feature**

Due to the relative simplicity of the limit orders feature, it's unlikely that use of redux for state management would be advantageous. Instead, we can probably maintain state in-place with use of react hooks and react-query, similar to the arbitrum bridge claims feature:
- react hooks for form state management (i.e asset selections, inputs, etc)
- react-query for fetching and async operations (e.g signing, getting a quote, etc)
- react-router `MemoryRouter` for UI routing

### UI Specifics

**TL;DR - side-by-side layout like trade quotes, otherwise refer to figma**

- New tab next to "Trade/Bridge" and "Claim", with route to go with it.
- Higher order component similar to `TradeInput`, called something like `LimitOrderInput`. This will consume the new share trade input component without swapper specifics.
- New input component nested alongside the above for configuring limit order params, called something like `LimitOrderConfig`.
- A side-by-side view of the `LimitOrderInput` and `LimitOrderHistory` is presented when screen width is wide enough (similar to trade quote list). Use the same approach as trade quote list, including animations, refs etc.
   - TBA on how this is presented or accessed on mobile/small screens.
